addons/point_of_sale/models/pos_session.py

```diff
@@ -1606,179 +1652,211 @@ def _pos_ui_models_to_load(self):
-        return {'domain': [('id', '=', self.config_id.picking_type_id.id)], 'fields': ['use_create_lots', 'use_existing_lots']}
+        return {
+            'search_params': {
+                'domain': [('id', '=', self.config_id.picking_type_id.id)],
+                'fields': ['use_create_lots', 'use_existing_lots'],
+            },
+        }
 
     def _get_pos_ui_stock_picking_type(self, params):
-        return self.env['stock.picking.type'].search_read(params['domain'], params['fields'])
+        return self.env['stock.picking.type'].search_read(**params['search_params'])[0]
 
     def _loader_params_res_users(self):
-        domain = [
-            ('company_ids', 'in', self.config_id.company_id.id),
-            '|',
-            ('groups_id', '=', self.config_id.group_pos_manager_id.id),
-            ('groups_id', '=', self.config_id.group_pos_user_id.id),
-        ]
-
-        return {'domain': domain, 'fields': ['name', 'company_id', 'id', 'groups_id', 'lang']}
+        return {
+            'search_params': {
+                'domain': [('id', '=', self.env.user.id)],
+                'fields': ['name', 'groups_id'],
+            },
+        }
 
     def _get_pos_ui_res_users(self, params):
-        return self.env['res.users'].search_read(params['domain'], params['fields'])
+        user = self.env['res.users'].search_read(**params['search_params'])[0]
+        user['role'] = 'manager' if any(id == self.config_id.group_pos_manager_id.id for id in user['groups_id']) else 'cashier'
+        del user['groups_id']
+        return user
 
     def _loader_params_product_pricelist(self):
         if self.config_id.use_pricelist:
             domain = [('id', 'in', self.config_id.available_pricelist_ids.ids)]
         else:
             domain = [('id', '=', self.config_id.pricelist_id.id)]
-
-        return {'domain': domain, 'fields': ['name', 'display_name', 'discount_policy']}
+        return {'search_params': {'domain': domain, 'fields': ['name', 'display_name', 'discount_policy']}}
 
     def _get_pos_ui_product_pricelist(self, params):
-        return self.env['product.pricelist'].search_read(params['domain'], params['fields'])
+        pricelists = self.env['product.pricelist'].search_read(**params['search_params'])
+        for pricelist in pricelists:
+            pricelist['items'] = []
+
+        pricelist_by_id = {pricelist['id']: pricelist for pricelist in pricelists}
+        pricelist_item_domain = [('pricelist_id', 'in', [p['id'] for p in pricelists])]
+        for item in self.env['product.pricelist.item'].search_read(pricelist_item_domain, []):
+            pricelist_by_id[item['pricelist_id'][0]]['items'].append(item)
+            if item.get('base_pricelist_id') and pricelist_by_id.get(item['base_pricelist_id'][0]):
+                item['base_pricelist'] = pricelist_by_id.get(item['base_pricelist_id'][0])
+            else:
+                item['base_pricelist'] = None
```

Comment:
I think we can write this better.
```suggestion
            item['base_pricelist'] = pricelist_by_id.get(item['base_pricelist_id'][0]) if item['base_pricelist_id'] else None
```